modules/input.nix

```diff
@@ -0,0 +1,386 @@
+        '';
+      };
+      tapAndDrag = mkOption {
+        type = with types; nullOr bool;
+        default = null;
+        example = true;
+        description = ''
+          Enables tap-and-drag for the touchpad.
+        '';
+      };
+      tapDragLock = mkOption {
+        type = with types; nullOr bool;
+        default = null;
+        example = true;
+        description = ''
+          Enables tap-and-drag lock for the touchpad.
+        '';
+      };
+      scrollMethod = mkOption {
+        type = with types; nullOr (enum (builtins.attrNames scrollMethods));
+        default = null;
+        example = "touchPadEdges";
+        description = ''
+          How scrolling is performed on the touchpad.
+        '';
+        apply = method: if (method == null) then null else scrollMethods."${method}";
+      };
+      rightClickMethod = mkOption {
+        type = with types; nullOr (enum (builtins.attrNames rightClickMethods));
+        default = null;
+        example = "twoFingers";
+        description = ''
+          How right-clicking is performed on the touchpad.
+        '';
+        apply = method: if (method == null) then null else rightClickMethods."${method}";
+      };
+      twoFingerTap = mkOption {
+        type = with types; nullOr (enum [ "rightClick" "middleClick" ]);
+        default = null;
+        example = "twoFingers";
+        description = ''
+          How right-clicking is performed on the touchpad.
+        '';
+        apply = v: if (v == null) then null else (v == "middleClick");
+      };
+    };
+  };
+  touchPadToConfig = touchpad:
+    let
+      touchName = touchpad.name;
+      touchVendor = touchpad.vendorId;
+      touchProduct = touchpad.productId;
+    in
+    {
+      "Libinput/${touchVendor}/${touchProduct}/${lib.escape ["/"] touchName}" = {
+        Enabled = touchpad.enable;
+        DisableWhileTyping = touchpad.disableWhileTyping;
+        LeftHanded = touchpad.leftHanded;
+        MiddleMouseEmulation = touchpad.middleMouseEmulation;
+        PointerAcceleration = touchpad.pointerSpeed;
+        NaturalScroll = touchpad.naturalScroll;
+        TapToClick = touchpad.tapToClick;
+        TapAndDrag = touchpad.tapAndDrag;
+        TapDragLock = touchpad.tapDragLock;
+        ScrollMethod = touchpad.scrollMethod;
+        ClickMethod = touchpad.rightClickMethod;
+        LmrTapButtonMap = touchpad.twoFingerTap;
+      };
+    };
+  
+  mouseType = types.submodule {
+    options = {
+      enable = mkOption {
+        type = with types; nullOr bool;
+        default = null;
+        example = true;
+        description = ''
+          Enables or disables the mouse
+        '';
+      };
+      name = mkOption {
+        type = types.str;
+        default = null;
+        example = "Logitech G403 HERO Gaming Mouse";
+        description = ''
+          The name of the mouse.
+          You can find it out running cat /proc/bus/input/devices | grep -B 1 -i mouse
```

Comment:
This doesn't work for me because my mouse doesn't have anything with mouse in the name. `grep '^H:.*mouse' -B 5 /proc/bus/input/devices` seems to work for me though. Seemingly this could maybe also be considered for the touchpad as well since I'm not sure all touchpads include `touchpad` in the name, but they should have `mouse*` in the handlers section

Reply:
![image](https://github.com/user-attachments/assets/91da7b16-75dd-4139-8b96-adc8e7ac58d9)
It works for me. But it is probably not a general rule.

Reply:
Your command works too. 
![image](https://github.com/user-attachments/assets/8320ad50-efe2-4804-9e5f-17e9c0629aba)

Reply:
Great. My suggestion to solve this would be to replace the old commands (in the three descriptions for both mouse and touchpad) with `grep '^H:.*mouse' -B 5 /proc/bus/input/devices`. We might be able to get away with the old command for touchpads, but afaik it's no requirement for those to have `touchpad` in the name either so I think the new command is still better for touchpads as well.

Reply:
For trackpads you can just grep  `/proc/bus/input/devices` since most trackpads are i2c devices and (afaik) i2c devices don't have names, so trackpads always have trackpad in the name.
For mice we could just check `lsusb`
![image](https://github.com/user-attachments/assets/6c41310f-79dc-4164-921c-1410f8cb1cbc)

Reply:
On my end when running the grep-command as it is now it matches `ETPS/2 Elantech Touchpad` which is in the name section of the touchpad in `/proc/bus/input/devices`, and if it was called something else it may not have matched. With that said I'm not sure if it's a real issue as most touchpads probably contain "touchpad" in some form in the name (unless it's called something with trackpad or something).
As for `lsusb` there I can see two potential problems, the first being that `lsusb` isn't installed on most nixos machines I don't think, so it's less accessible than just using `/proc/bus/input/devices`, and second it may not support bluetooth mice. The command provided above does just use `/proc/bus/input/devices`

Reply:
While I don't think your first problem is that big of a deal (you can use comma or `nix run`), I do agree with the second.
lsusb only lists usb devices, so it won't catch anything else. For now I would say to leave both, but we could consider making a python script that figures it out for you

Reply:
Yeah figuring this out automatically could be done through some startup script or something and is a task for later. My point was that the descriptions for the options should be correct/helpful. For that the grep command above would suffice for now I reckon, or alternatively we could just add "see /proc/bus/input/devices for the values" or something. A.t.m. the grep commands in the descriptions for the mice doesn't work and the grep commands for the touchpad only works for some hardware (only when the name contains touchpad), and I think this should be correct before we merge into `trunk`.

Reply:
I think a more general description like `see /proc/bus/input/devices for the values` would be better